ipykernel/kernelbase.py

```diff
@@ -340,6 +344,22 @@ def send_response(self, stream, msg_or_type, content=None, ident=None,
         """
         return self.session.send(stream, msg_or_type, content, self._parent_header,
                                  ident, buffers, track, header, metadata)
+    
+    def init_metadata(self, parent):
+        """Initialize metadata.
+        
+        Run at the beginning of request handlers.
```

Comment:
@minrk this comment didn't get a response, in case you missed it when reviewing.